src/gpu/nvidia/cudnn_batch_normalization_executor.hpp

```diff
@@ -40,155 +41,161 @@ struct bnorm_exec_base_t {
-            auto dx = sc.memory<void *>(ih, diff_src_acc);
-            auto scale = sc.memory<uint8_t *>(ih, scale_acc);
-            auto bias = sc.memory<uint8_t *>(ih, bias_acc)
-                    + (bnorm_impl->C() * sizeof(float));
-            auto diff_scale = sc.memory<uint8_t *>(ih, diff_scaleshift_acc);
-            auto diff_bias = diff_scale + (bnorm_impl->C() * sizeof(float));
-            auto save_mean = sc.memory<uint8_t *>(ih, wkspace_acc);
-            auto save_var = save_mean + bnorm_impl->mean_var_size_bytes();
-            auto wkspace = save_var + bnorm_impl->mean_var_size_bytes();
-            auto relu_dy = bnorm_impl->fuse_norm_relu()
+            if (init_ss)
+                init_scaleshift(
+                        sc, ih, cuda_stream, arg_scale_bias, bnorm_impl->C());
+
+            auto *x = arg_src.get_native_pointer(ih, sc);
+            auto *dy = arg_diff_dst.get_native_pointer(ih, sc);
+            auto *dx = arg_diff_src.get_native_pointer(ih, sc);
+
+            auto *scale = static_cast<uint8_t *>(
+                    arg_scale_bias.get_native_pointer(ih, sc));
+            auto *bias = scale + bnorm_impl->C() * sizeof(float);
+
+            auto *diff_scale = static_cast<uint8_t *>(
+                    arg_diff_scale_bias.get_native_pointer(ih, sc));
+            auto *diff_bias = diff_scale + (bnorm_impl->C() * sizeof(float));
+
+            auto *save_mean = static_cast<uint8_t *>(
+                    arg_wkspace.get_native_pointer(ih, sc));
+            auto *save_var = save_mean + bnorm_impl->mean_var_size_bytes();
+            auto *wkspace = save_var + bnorm_impl->mean_var_size_bytes();
+            auto *relu_dy = bnorm_impl->fuse_norm_relu()
                     ? sc.memory<void *>(ih, *temp_relu_output)
                     : nullptr;
 
             std::shared_ptr<bnorm_args_t> args(
                     new bnorm_bwd_args_t(x, dx, dy, save_mean, save_var, scale,
                             bias, diff_scale, diff_bias, wkspace, relu_dy));
-
             bnorm_impl->execute(handle, args);
         });
     }
 
-    template <typename T>
-    void maybe_init_ss(
-            nvidia::sycl_cuda_stream_t *cuda_stream, T, T, bool) const {}
-
-    template <typename T>
-    void maybe_init_ss(nvidia::sycl_cuda_stream_t *cuda_stream,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> scale_acc,
-            ::sycl::accessor<T, 1, ::sycl::access::mode::write> bias_acc,
-            bool init_ss) const {
-        if (init_ss) {
-            constexpr T scale_val = 1, bias_val = 0;
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(scale_acc, scale_val);
-            });
-
-            cuda_stream->interop_task([&](::sycl::handler &cgh) {
-                cgh.fill(bias_acc, bias_val);
-            });
-        }
+    template <typename T = float>
+    void init_scaleshift(cuda_sycl_scoped_context_handler_t &sc,
```

Comment:
What is the reason for having this empty function?

Reply:
There are cases where ```init_scaleshift``` is called for ```nullptr``` or ```read_acc_t``` types. This empty function allows for such cases. This is the same design as the existing implementation

Reply:
I see, thanks for the explanation.